test/unit/helper/StateWidgetTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoResourceWorkflow\test\unit\helper;
+
+use core_kernel_classes_Resource;
+use oat\oatbox\service\ServiceManager;
+use oat\tao\model\service\ApplicationService;
+use oat\taoResourceWorkflow\helper\StateWidget;
+use oat\taoResourceWorkflow\model\ResourceWorkflowService;
+use oat\taoResourceWorkflow\model\wfmodel\JsonWorkflow;
+use oat\taoResourceWorkflow\model\wfmodel\StateObject;
+use oat\taoResourceWorkflow\model\wfmodel\TransitionObject;
+use oat\taoResourceWorkflow\model\WorkflowModel;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class StateWidgetTest extends TestCase
+{
+    public const ITEM_URI = 'ItemName';
+    public const RESOURCE_TYPE_KEY = 'resourceType';
+    public const RDF_ITEM_URI = 'http://www.tao.lu/Ontologies/TAOItem.rdf#Item';
+    private const STATE_LABEL_1 = 'label_1';
+    public const STATE_LABEL_2 = 'label_2';
+    private const TRANSITION_URL_1 = 'url_1';
+    private const TRANSITION_URL_2 = 'url_2';
+    private const TRANSITION_ID_1 = 'id_1';
+    private const TRANSITION_ID_2 = 'id_2';
+
+    /** @var JsonWorkflow|MockObject  */
+    private $workflowModelMock;
+
+    /** @var ResourceWorkflowService|MockObject */
+    private $resourceWorkflowServiceMock;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (!defined('ROOT_URL')) {
+            define('ROOT_URL', __DIR__ . '/../../../../../');
+        }
+        if (!defined('CONFIG_PATH')) {
+            define('CONFIG_PATH', ROOT_URL . 'config/');
+        }
+        if (!defined('EXTENSION_PATH')) {
+            define('EXTENSION_PATH', ROOT_URL);
+        }
```

Comment:
Why do you need this if the phpunit.xml is using `generis/test/bootstrap.php` which contains all these values declared?

Reply:
```suggestion
```